cluster/src/main/scala/io/snappydata/ToolsCallbackImpl.scala

```diff
@@ -47,4 +53,34 @@ object ToolsCallbackImpl extends ToolsCallback {
       classLoader: ClassLoader): Unit = {
     SnappyUtils.setSessionDependencies(sparkContext, appName, classLoader)
   }
+
+  override def addURIs(alias: String, jars: Array[String], deploySql: String): Unit = {
+    if (alias != null) {
+      Misc.getMemStore.getGlobalCmdRgn.put(alias, deploySql)
+    }
+    val lead = ServiceManager.getLeadInstance.asInstanceOf[LeadImpl]
+    val loader = lead.urlclassloader
+    jars.foreach(j => {
+      val url = new File(j).toURI.toURL
+      loader.addURL(url)
+    })
+  }
+
+  override def addURIsToExecutorClassLoader(jars: Array[String]): Unit = {
+    if (ExecutorInitiator.snappyExecBackend != null) {
+      val snappyexecutor = ExecutorInitiator.snappyExecBackend.executor.asInstanceOf[SnappyExecutor]
+      snappyexecutor.updateMainLoader(jars)
+    }
+  }
+
+  override def getAllGlobalCmnds(): Array[String] = {
+    GemFireXDUtils.waitForNodeInitialization()
+    Misc.getMemStore.getGlobalCmdRgn.values().toArray.map(_.asInstanceOf[String])
+  }
+
+  override def removePackage(alias: String): Unit = {
```

Comment:
Can you please explain here a bit, how remove package would work? What's the scope of it? I believe it's only removing from the region and not from the actual classloader. Should we not change lead & executors classloaders here?